### ISSUE TITLE
ggshield pre-receive scans more commits than necessary

### ISSUE THREAD

**Issue by agateau-gg** [GitGuardian/ggshield#369]
## Environment

- ggshield version: 1.13.3
- Operating system (Linux, macOS, Windows): Linux
- Operating system version: Ubuntu 20.04
- Python version: 3.8

## Describe the bug

This is the pre-receive variant of #303.

When scanning a new branch, ggshield pre-receive hook is not able to detect the branch start and resort to scanning the last N commits (N being 50 by default).

Steps to reproduce:

1. Setup a remote repository
2. Install ggshield pre-receive hook on it
3. Clone it locally
4. Create a new branch in the local clone
5. Push the new branch

Actual result:

The last 50 commits are scanned.

Expected result:

Only the new commits should be scanned.

**Comment by donald-ngo** [comment]
Getting this error after doing:
```
~/Source/Gitlab-Local/local-poc-service-sentinel-prod [main] $ git checkout -b donald3
Switched to a new branch 'donald3'
~/Source/Gitlab-Local/local-poc-service-sentinel-prod [donald3] $ git push -u origin donald3
```


```
Traceback (most recent call last):
  File "/var/opt/gitlab/.local/bin/ggshield", line 8, in <module>
    sys.exit(main())
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/ggshield/cmd/main.py", line 180, in main
    return cli.main(args, prog_name="ggshield", standalone_mode=not show_crash_log)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/ggshield/cmd/secret/scan/prereceive.py", line 138, in prereceive_cmd
    before, after = parse_stdin()
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/ggshield/cmd/secret/scan/prereceive.py", line 102, in parse_stdin
    start_commit = find_branch_start(new_commit)
  File "/var/opt/gitlab/.local/lib/python3.8/site-packages/ggshield/cmd/secret/scan/prereceive.py", line 84, in find_branch_start
    return ancestors[0]
```

**Comment by agateau-gg** [comment]
Should be fixed by #387
